### PR TITLE
boost: bump deps

### PR DESCRIPTION
**Pull request by robomics**
Specify library name and version:  **boost/all**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@robomics Sorry the delay. We are avoiding bump dependencies at first sight as they could result in version conflicts. Boost is not only well used in CCI, but its dependencies are also used in many other recipes. Do you need this bump dependencies to fix a conflict, or you are trying to use the latest version only of those libraries?

We are also working on trying to use version ranges for more projects. I'm taking a look if we can use zstd as well, so it could help to avoid conflicts in the future.

File: recipes/boost/all/conanfile.py [hunk 1]
```diff
@@ -651,14 +651,14 @@ def requirements(self):
         if self._with_bzip2:
             self.requires("bzip2/1.0.8")
         if self._with_lzma:
-            self.requires("xz_utils/5.4.4")
+            self.requires("xz_utils/5.4.5")
```
Comment (valgur) [inline recipes/boost/all/conanfile.py:654]: +1 for using `5.4.5`, which is the version used in nearly all other recipes except ones still awaiting migration to Conan 2.0.

**Comment by robomics** [comment]
Thanks for looking into this @uilianries!

When I submitted this PR I was declaring dependencies using `conanfile.txt` in several of my projects, and so this PR was an attempt to address various version conflicts I was facing.
In the mean time I've switched to using `conanfile.py` files, so that I can force/override dependency versions as needed.

Let me know if i should push a commit to use version ranges for zstd and v5.4.5 for xz_utils, as suggested by valgur.

**Review by uilianries — APPROVED** [review]
@robomics Sorry for blocking the PR, it was intended to avoid merging it by mistake. 

After talking to Conan team, the number of affected packages is small and we can open separated PRs to fix as well. At time, Boost is largely used in CCI and should continue working with your change.

I updated your PR with few changes:

- Use version range for zstd. Still, it's the first recipe using it.
- Add version range for xz_utils. Not documented yet, but looks very stable and compatible. 

Hope these changes fix your current conflicts.

**Comment by robomics** [comment]
Awesome! Many thanks!

**Review by valgur — APPROVED** [review]

**Pull request merged**